src/nvim/window.c

```diff
@@ -6086,20 +6489,26 @@ void win_findbuf(typval_T *argvars, list_T *list)
 
 void win_ui_flush(void)
 {
-  if (!ui_is_external(kUIMultigrid)) {
+  if (!ui_is_external(kUIMultigrid) && !ui_is_external(kUIFloat)) {
     return;
   }
 
   FOR_ALL_TAB_WINDOWS(tp, wp) {
     if (wp->w_pos_changed && wp->w_grid.chars != NULL) {
       if (tp == curtab) {
-        ui_call_win_pos(wp->w_grid.handle, wp->handle, wp->w_winrow,
-                        wp->w_wincol, wp->w_width, wp->w_height);
+        ui_ext_win_position(wp);
       } else {
         ui_call_win_hide(wp->w_grid.handle);
       }
       wp->w_pos_changed = false;
     }
   }
+}
 
+win_T *lastwin_nofloating(void) {
+  win_T *res = lastwin;
+  while (res->w_floating) {
+    res = res->w_prev;
```

Comment:
Maybe it's worth  a sanity check:

    assert(res != NULL && res != firstwin);